src/components/table/SnapshotAccessRequestTable.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Dispatch } from 'react';
+import { SnapshotAccessRequestResponse } from 'generated/tdr';
+import { TableColumnType } from 'reducers/query';
+import LightTable from 'components/table/LightTable';
+import moment from 'moment/moment';
+import { Button } from '@mui/material';
+import _ from 'lodash';
+import { rejectSnapshotAccessRequest } from 'actions';
+import { Action } from 'redux';
+
+interface IProps {
+  dispatch: Dispatch<Action>;
+  snapshotAccessRequests: Array<SnapshotAccessRequestResponse>;
+  loading: boolean;
+  searchString: string;
+  refreshCnt: number;
+}
+
+function SnapshotAccessRequestTable({
+  dispatch,
+  snapshotAccessRequests,
+  loading,
+  searchString,
+  refreshCnt,
+}: IProps) {
+  const containsSearchString = (request: SnapshotAccessRequestResponse): boolean =>
+    _.includes(request.id, searchString) ||
+    _.includes(request.snapshotName, searchString) ||
+    _.includes(request.sourceSnapshotId, searchString) ||
+    _.includes(request.createdBy, searchString);
+
+  const filteredRequests = _.filter(
+    snapshotAccessRequests,
+    (snapshotAccessRequest: SnapshotAccessRequestResponse) =>
+      containsSearchString(snapshotAccessRequest),
+  );
+
+  const filteredAndSortedRequests = _.sortBy(
+    filteredRequests,
+    (snapshotAccessRequest: SnapshotAccessRequestResponse) =>
+      Date.parse(
+        snapshotAccessRequest.createdDate ? snapshotAccessRequest.createdDate : _.now().toString(),
+      ) * -1,
+  );
```

Comment:
Would it make sense to put each of these in a useeffect?

Reply:
Because there is no async and very limited cost, I think we can safely have them outside of a `useEffect`. The advantage of having them as derived values that are inlined, as opposed to within a `useEffect`, is that we don't need to manage them as state - the scoping is handled for us.

Reply:
The time when you really want to `useEffect` is when you want to store values in state, because you don't want a `setState` call within the render loop.